tests/storage/export.go

```diff
@@ -1455,52 +1469,151 @@ var _ = SIGDescribe("Export", func() {
+							},
+							Max: k8sv1.ResourceList{
+								k8sv1.ResourceMemory: resource.MustParse("2Gi"),
+								k8sv1.ResourceCPU:    resource.MustParse("2"),
+							},
+							Min: k8sv1.ResourceList{
+								k8sv1.ResourceMemory: resource.MustParse("1Mi"),
+								k8sv1.ResourceCPU:    resource.MustParse("1m"),
+							},
+						},
+					},
+				},
+			}
+			lr, err = virtClient.CoreV1().LimitRanges(namespace).Create(context.Background(), lr, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			return vmi.Status.Phase
-		}, 180*time.Second, time.Second).Should(Equal(virtv1.Running))
-		// For testing the token is the name of the source VM.
-		token := createExportTokenSecret(vmi.Name, vmi.Namespace)
-		pvcName := ""
-		for _, volume := range vmi.Spec.Volumes {
-			if volume.DataVolume != nil {
-				pvcName = volume.DataVolume.Name
+			By("Ensuring LimitRange exists")
+			Eventually(func() error {
+				lr, err = virtClient.CoreV1().LimitRanges(namespace).Get(context.Background(), lr.Name, metav1.GetOptions{})
+				return err
+			}, 30*time.Second, 1*time.Second).Should(BeNil())
+		}
+
+		removeLimitRangeFromNamespace := func() {
+			if lr != nil {
+				err = virtClient.CoreV1().LimitRanges(lr.Namespace).Delete(context.Background(), lr.Name, metav1.DeleteOptions{})
+				if !errors.IsNotFound(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+				lr = nil
 			}
+			tests.UpdateKubeVirtConfigValueAndWait(originalConfig)
 		}
-		Expect(pvcName).ToNot(BeEmpty())
-		export := createPVCExportObject(pvcName, vmi.Namespace, token)
-		Expect(export).ToNot(BeNil())
-		waitForExportPhase(export, exportv1.Pending)
-		waitForExportCondition(export, expectedPVCInUseCondition(dataVolume.Name, dataVolume.Namespace), "export should report pvc in use")
 
-		By("Deleting VMI, we should get the export ready eventually")
-		deleteVMI(vmi)
-		export = waitForReadyExport(export)
-		checkExportSecretRef(export)
-		Expect(*export.Status.TokenSecretRef).To(Equal(token.Name))
-		verifyKubevirtInternal(export, export.Name, export.Namespace, vmi.Spec.Volumes[0].DataVolume.Name)
-		By("Starting VMI, the export should return to pending")
-		vmi = tests.NewRandomVMIWithDataVolume(dataVolume.Name)
-		vmi = createVMI(vmi)
-		waitForExportPhase(export, exportv1.Pending)
-		waitForExportCondition(export, expectedPVCInUseCondition(dataVolume.Name, dataVolume.Namespace), "export should report pvc in use")
+		BeforeEach(func() {
+			originalConfig = *util.GetCurrentKv(virtClient).Spec.Configuration.DeepCopy()
+		})
+
+		AfterEach(func() {
+			removeLimitRangeFromNamespace()
+		})
+
+		It("[Serial] should report export pending if PVC is in use because of VMI using it, and start the VM export if the PVC is not in use, then stop again once pvc in use again", Serial, func() {
+			sc, exists := libstorage.GetRWOFileSystemStorageClass()
+			if !exists {
+				Skip("Skip test when Filesystem storage is not present")
```

Comment:
Can we use decorators instead of a skip?

Reply:
It was like that, but if you have an example of a decorator I can certainly look at changing it.

Reply:
As discussed, this is out of scope of this PR, but opening a GH issue to look at replacing the skips with decorators.